__tests__/changelog/utils/generetaChangelog.test.ts

```diff
@@ -0,0 +1,230 @@
+      2,
+      'v1.2.3',
+    );
+    expect(gitlab.fetchProjectCommitsSince as jest.Mock).toHaveBeenCalledTimes(
+      1,
+    );
+    // second arg is ISO string greater than tag created_at by ~1s
+    const sinceArg = (gitlab.fetchProjectCommitsSince as jest.Mock).mock
+      .calls[0][1];
+    expect(typeof sinceArg).toBe('string');
+
+    expect(result).toBe(
+      '- fix: bug - [XYZ-7](https://tickets.example.com/XYZ-7)',
+    );
+  });
+
+  it('maps merge commits (squash) to MR link with ticket', async () => {
+    const mergeCommit = baseCommit({
+      id: 'm1',
+      short_id: 'm1',
+      title: "Merge branch 'feature' into 'main'",
+      message:
+        "Merge branch 'feature' into 'main'\n\nSee merge request group/proj!42",
+      created_at: '2025-01-03T09:00:00Z',
+    });
+
+    (gitlab.fetchProjectCommits as jest.Mock).mockResolvedValue([
+      mergeCommit,
+      baseCommit({ id: 'c4', short_id: 'c4' }),
+    ]);
+
+    (gitlab.fetchMergeRequestByIid as jest.Mock).mockResolvedValue({
+      web_url: 'https://gitlab.example.com/group/proj/-/merge_requests/42',
+      squash: true,
+      squash_commit_sha: 'squashsha',
+    });
+
+    (gitlab.fetchProjectCommit as jest.Mock).mockResolvedValue(
+      baseCommit({
+        id: 'squashsha',
+        short_id: 'squashsha',
+        title: 'feat: implement super thing',
+        message: 'feat: implement super thing ABC-9',
+        created_at: '2025-01-03T09:00:01Z',
+      }),
+    );
+
+    const result = await generateChangelog(3, undefined);
+
+    expect(gitlab.fetchMergeRequestByIid as jest.Mock).toHaveBeenCalledWith(
+      3,
+      42,
+    );
+    expect(gitlab.fetchProjectCommit as jest.Mock).toHaveBeenCalledWith(
+      3,
+      'squashsha',
+    );
+    expect(result).toBe(
+      '- [feat: implement super thing](https://gitlab.example.com/group/proj/-/merge_requests/42) - [ABC-9](https://tickets.example.com/ABC-9)',
+    );
+  });
+
+  it('removes duplicated commits keeping the oldest created_at', async () => {
+    const mergeCommit1 = baseCommit({
+      id: 'mA',
+      short_id: 'mA',
+      title: "Merge branch 'feature' into 'main'",
+      message:
+        "Merge branch 'feature' into 'main'\n\nSee merge request group/proj!10",
+      created_at: '2025-01-04T10:00:00Z',
+    });
+    const mergeCommit2 = baseCommit({
+      id: 'mB',
+      short_id: 'mB',
+      title: "Merge branch 'feature' into 'release'",
+      message:
+        "Merge branch 'feature' into 'release'\n\nSee merge request group/proj!11",
+      created_at: '2025-01-04T11:00:00Z',
+    });
+
+    (gitlab.fetchProjectCommits as jest.Mock).mockResolvedValue([
+      mergeCommit1,
+      mergeCommit2,
+    ]);
+
+    // Both MRs point to commits with the same message but different created_at
+    (gitlab.fetchMergeRequestByIid as jest.Mock)
+      .mockResolvedValueOnce({
+        web_url: 'https://gitlab.example.com/.../mr/10',
```

Comment:
Suggestion: can you put a fake slug instead of ... ?

Reply:
Done